### PR TITLE
Added support for temporary table upload in `alma`

### PR DESCRIPTION
**Pull request by andamian**
Solution for #3030 

Also cleaned up some of the alma remote tests to make them a little bit faster.

### CODE REVIEW

File: astroquery/alma/core.py [hunk 1]
```diff
@@ -678,19 +678,28 @@ def query_sia(self, *, pos=None, band=None, time=None, pol=None,
...
+            ADQL query to execute
         maxrec : int
             maximum number of records to return
+        uploads : dict
+            a mapping from temporary table names to objects containing a votable. These
+            temporary tables can be referred to in queries. The keys in the dictionsry are
+            the names of temporary tables which need to be prefixed with the TAP_UPLOAD
+            schema in the actual query. The values are either astropy.table.Table instances
+            or file names or file like handles such as io.StringIO to table definition in
+            IVOA VOTable format. Ex: uploads={'tmptable': '/tmp/tmptable_def.xml'}
```
Comment (keflavich) [inline astroquery/alma/core.py:698]:
```suggestion
            IVOA VOTable format. 
            
    Examples
    ----------
    >>> uploads = {'tmptable': '/tmp/tmptable_def.xml'}
    >>> rslt = query_tap(self, query, maxrec=None, uploads=uploads)
```

**Review by keflavich — COMMENTED** [review]
Looks like a nice addition, just needs a little clearer documentation - can you provide an example upload table in the narrative docs?

**Comment by andamian** [comment]
Not sure what the problem is now. There's 100% patch coverage.

**Comment by bsipocz** [comment]
> Not sure what the problem is now. There's 100% patch coverage.

Ignore that, coverage number doesn't take remote tests into account.

File: astroquery/alma/core.py [hunk 2]
```diff
@@ -678,19 +678,33 @@ def query_sia(self, *, pos=None, band=None, time=None, pol=None,
 
     query_sia.__doc__ = query_sia.__doc__.replace('_SIA2_PARAMETERS', SIA2_PARAMETERS_DESC)
 
-    def query_tap(self, query, maxrec=None):
+    def query_tap(self, query, maxrec=None, uploads=None):
```
Comment (bsipocz) [inline astroquery/alma/core.py:681]: Can we make things kwarg only?

```suggestion
    def query_tap(self, query, *, maxrec=None, uploads=None):
```

**Review by bsipocz — COMMENTED** [review]
Overall looks good to me. I'll leave the actual approval and merge to Adam though.

One minor comment, that I see a bunch of test failures in the docs, all of those are present on `main`, too but it would be nice to eventually clean them up.

**Comment by andamian** [comment]
Can this be merged or do I need to try to fix the failing check (which I have no idea why is failing)? Thanks

**Comment by bsipocz** [comment]
No, coverage is a red herring (it doesn't take into account the remote coverage and anyways likes to act up).

I was waiting to give Adam a chance to chime in, but I suppose it's time to get this merged. Thanks!

**Review by bsipocz — APPROVED** [review]

**Comment by keflavich** [comment]
Yep, sorry, I was happy with this.

**Comment by andamian** [comment]
Thank you both.

**Pull request merged**